proto/src/util.rs

```diff
@@ -36,3 +38,53 @@ pub fn create_test_channel_credentials() -> ChannelCredentials {
         .root_cert(ca.into())
         .build()
 }
+
+impl TryFrom<grpcio::RpcStatus> for Status {
+    type Error = grpcio::Error;
+
+    fn try_from(value: grpcio::RpcStatus) -> grpcio::Result<Self> {
+        let mut s = Status::default();
+        #[cfg(feature = "protobuf-codec")]
+        protobuf::Message::merge_from_bytes(&mut s, value.error_details())?;
+        #[cfg(feature = "prost-codec")]
+        prost::Message::merge(&mut s, value.error_details())?;
+        if s.code == value.error_code().into() {
+            if s.message == value.error_message() {
```

Comment:
This seems like a temporary check that doesn't need to stay in when we are confident the new version is working?

Reply:
The library doesn't know whether the other end, which can be other library or even in different language, is implemented correctly.

Reply:
Still I am wondering if in production this is the best behavior? It seems like something where I would prefer to log a warning but return Ok since the code is still correct.

Reply:
Then there is no way to detect the error. If user want to ignore the error, they can still parse the status from `error_details` directly.